### PR TITLE
Promote schema-level repetition parameters to plain create_schema() arguments

### PR DESCRIPTION
**Pull request by Felixmil**
Closes #167

This PR promotes the three schema-level repetition parameters of an Advanced Protocol schema to plain, unit-aware arguments on `create_schema()`. Previously a caller had to supply `NumberOfRepetitions`, `TimeBetweenRepetitions`, and `Start time` through the free-form `parameters` list, spelling PK-Sim's internal names verbatim and with no unit validation on the time values. These three parameters appear on every schema-bearing model in the corpus, so they are now first-class arguments that follow the scalar validation shape of `create_output_interval()` and the conflict-error house style of `create_protocol()`. The `parameters` list stays as the untouched escape hatch for anything else, and every existing call is unaffected.

## Changes

- `create_schema()` gains, after the existing `name`, `parameters`, `items`, the arguments `number_of_repetitions = NULL`, `time_between_repetitions = NULL`, `time_between_repetitions_unit = "h"`, `start_time = NULL`, `start_time_unit = "h"`. Each time value is immediately followed by its unit companion; `number_of_repetitions` is dimensionless and carries no unit.
- Each supplied promoted value is validated as a single finite numeric (mirroring `create_output_interval()`), factored into a small internal `check_finite_scalar()` helper placed after the exported function. `number_of_repetitions` additionally requires a whole number, checked with a tolerance (`abs(x - round(x)) < .Machine$double.eps^0.5`) so a non-scalar or non-finite value hits the single-finite-numeric message before the whole-number branch.
- A time value's unit is validated with `validate_unit(unit, "Time")` only inside the `if (!is.null(value))` block, so a `NULL` time value never validates its unit.
- Promoted entries are built with `create_parameter()` using the exact PK-Sim `Name` strings (`"NumberOfRepetitions"`, `"TimeBetweenRepetitions"`, `"Start time"`, including the space); the time entries keep their `Unit` key while `NumberOfRepetitions` has none.
- Conflict detection runs before the merge: it resolves each `parameters` entry's name from `Name`, falling back to `Path`, and exact-matches against the pksim names of the supplied promoted arguments. Supplying a value both as a promoted argument and in `parameters` aborts once with a header, an `"i"` bullet, and an `"x"` bullet listing all offenders, in `create_protocol()`'s style. A matching `parameters` entry with no corresponding promoted argument is not a conflict.
- Merge order: existing `parameters` entries first, then promoted entries in signature order. When neither is supplied, `data$Parameters` is left unset (byte-identical to before); when only promoted arguments are supplied, it is built from those alone.
- Roxygen documents the five new arguments, notes that `parameters` remains the escape hatch and each promoted argument is mutually exclusive with its matching entry, and adds an `@examples` block using the plain-argument form alongside the existing `parameters`-list example. `man/create_schema.Rd` was regenerated with `devtools::document()`.
- `NEWS.md` gains one user-facing bullet under the development-version heading, ordered alphabetically before the `create_snapshot()` bullet.
- `tests/testthat/test-create_schema.R` adds self-contained blocks covering AC-1 through AC-15 (correct `Name`/`Value`/`Unit`, absence of `Unit` for `NumberOfRepetitions`, default and custom time units, invalid-unit error, whole-number error, non-scalar and `NA` errors, a conflict for each of the three names including a `Path`-based one, a single multi-conflict error, no false positive, and merge ordering). Error cases use `expect_snapshot(error = TRUE)` and raw-entry assertions use `expect_snapshot()`. The existing tests are unchanged and still cover backward compatibility.

## Verification

All commands were run from the package root with `NOT_CRAN=true`.

- `devtools::document()`: regenerated `man/create_schema.Rd` with the new arguments; exited cleanly. The only warnings are pre-existing `Snapshot` topic warnings that are present on the base branch and unrelated to this change.
- `devtools::test(filter = "create_schema")`: after recording the new snapshots, the file is green (FAIL 0, WARN 0, SKIP 0, PASS 59).
- `devtools::test()` (full suite): PASS 3128, SKIP 1, FAIL 3. The 3 failures are all in `test-snapshot.R` (snapshot version-ceiling tests) and were confirmed to pre-exist on the base branch with these changes stashed, so they are unrelated to this change.

# QA: Approved

The implementation faithfully promotes the three schema-level repetition parameters on `create_schema()` exactly as the spec and plan prescribe: correct signature order, correct pksim `Name` strings (including the space in `"Start time"`), gated unit validation, tolerance-based whole-number check, all-offenders-at-once conflict detection resolving `Path`, backward-compatible merge, regenerated `man/`, a single correct `NEWS.md` bullet, and tests plus recorded snapshots covering AC-1 through AC-15. No correctness defects or convention violations were found.

<details>
<summary>Full QA report</summary>

## Method

Read `spec.md`, `plan.md`, `build.md`; the PR diff (`gh pr diff 175`); PR metadata; the post-change `R/create_schema.R`, `NEWS.md`, `man/create_schema.Rd`, `tests/testthat/test-create_schema.R`, and `tests/testthat/_snaps/create_schema.md`; plus supporting sources `R/Parameter.R`, `R/LocalizedParameter.R`, `R/utils.R`, `R/utils-create.R` to confirm the helpers the diff relies on behave as assumed. Ran two lenses: adversarial (what is missing/overstated) and edge-case hunting (path-bearing conflicts, `NA`, vector inputs, byte-identity).

## Requirement-to-evidence mapping

**AC-1 (number_of_repetitions, no Unit) — pass.** `R/create_schema.R:105-108` builds `create_parameter(name = "NumberOfRepetitions", value = ...)` with no `unit`; `R/Parameter.R:365,369-371` confirms no `Unit` key is added when `unit` is `NULL`. Test at `tests/testthat/test-create_schema.R:103-111` asserts `expect_named(entry, c("Name","Value"))` + `expect_null(entry$Unit)`; snapshot in `_snaps/create_schema.md` shows exactly `Name`/`Value`, no `Unit`.

**AC-2 / AC-3 / AC-4 (time entries with Unit, default and custom) — pass.** `R/create_schema.R:116-120` (TimeBetweenRepetitions, default `"h"`), `:128-132` (Start time). Snapshots confirm `Unit = "h"`, `Unit = "day(s)"`, and `{Name="Start time", Value=0, Unit="h"}`. The exact pksim `Name` strings, including the space in `"Start time"`, are present verbatim.

**AC-5 / FR-9 (merge order) — pass.** `R/create_schema.R:169-183`: existing `parameters` first (`to_raw_parameters(parameters, "Name")`), promoted appended after in signature order. Test at `:139-166` asserts the ordered `Name` vector `Foo, Bar, NumberOfRepetitions, TimeBetweenRepetitions, Start time`.

**AC-6 (invalid time unit) — pass.** `validate_unit(time_between_repetitions_unit, "Time")` at `R/create_schema.R:113`, inside the `if (!is.null(time_between_repetitions))` block. Snapshot shows the `validate_unit()` invalid-unit message.

**FR-5 (unit validated only when value supplied) — pass.** Both `validate_unit` calls (`:113`, `:125`) live strictly inside the respective `if (!is.null(value))` blocks, so a `NULL` time value with a bogus unit never validates. Correct by inspection.

**AC-7 (whole-number) / FR-6 — pass.** `R/create_schema.R:95-102` uses `abs(x - round(x)) >= .Machine$double.eps^0.5` (tolerance, not `is.integer()`); the whole-number check runs after `check_finite_scalar()` (`:94`), so a non-scalar/non-finite value never reaches `round()`. Snapshot shows "must be a single finite whole number".

**AC-8 (non-scalar / NA) — pass.** `check_finite_scalar` (`R/create_schema.R:195-203`) rejects `!is.numeric || length != 1 || !is.finite`. `c(1,2)` fails length; `NA` (logical) fails `is.numeric`. Both hit the single-finite-numeric message before the whole-number branch. Two snapshots confirm the wording and the `create_schema()` call attribution (via `call = parent.frame()`).

**AC-9/10/11 (conflict per name) — pass.** `R/create_schema.R:140-162`. Snapshots show the three-bullet house-style error (header + `"i"` + `"x"`) naming `NumberOfRepetitions`, `TimeBetweenRepetitions`, and `Start time` (with the space) respectively.

**AC-12 (path-bearing conflict) / FR-7 — pass.** `create_parameter(path = "Start time")` returns a `LocalizedParameter` (`R/Parameter.R:429-431`), which `inherit`s from `Parameter` (`R/LocalizedParameter.R:35`), so `inherits(param, "Parameter")` at `R/create_schema.R:149` is `TRUE`, and `$data` yields `list(Path = "Start time", ...)`; `raw$Name %||% raw$Path` resolves to `"Start time"`. Snapshot confirms the conflict fires. This edge case is handled correctly.

**AC-13 (all conflicts in one error) — pass.** `intersect(promoted_names, existing_names)` at `R/create_schema.R:154` preserves signature order and dedupes; a single `cli_abort`. Snapshot lists all three names with correct cli pluralization ("Conflicting parameters: ... and ...").

**AC-15 (no false positive) / FR-7 — pass.** Conflict block is guarded by `length(promoted_names) > 0` (`:145`); when no promoted arg is supplied, `promoted_names` is empty and detection is skipped. Test at `tests/testthat/test-create_schema.R:263-273` supplies a `"NumberOfRepetitions"` parameters entry with no promoted arg and asserts it is preserved.

**AC-14 / FR-10 (backward compatibility, byte-identity) — pass.** When neither promoted arg is supplied, `promoted` is empty, so `R/create_schema.R:175-180` is skipped and `raw_parameters` is exactly `to_raw_parameters(parameters, "Name")` (`:173`), identical to the pre-change path; when neither `parameters` nor promoted is supplied, `raw_parameters` stays `NULL` and `data$Parameters` is never set. The `parameters must be a list` guard (`:86-88`) still fires before conflict detection. The existing test blocks and the existing snapshot block are unchanged (the diff to the test file and snapshot file is additive only, no deletions).

**AC-16 / FR-11 / FR-12 (docs, man, NEWS) — pass.** Roxygen (`R/create_schema.R:14-56` region in the diff) documents all five new args, notes mutual exclusivity, reframes `@param parameters` as the escape hatch, and adds an `@examples` block for the plain-arg form while keeping the `parameters`-list example. `man/create_schema.Rd` is regenerated (usage block and all five `\item`s match the roxygen; consistent with `devtools::document()` output, not hand-edited). `NEWS.md:3` has one single-line, present-tense bullet, `create_schema()` first, ending `(#167)`, sorted first in the development-version section (`create_schema` < `create_snapshot` alphabetically).

**FR-1 (signature order) — pass.** `R/create_schema.R:75-84` appends the five new args after `name`, `parameters`, `items`, each time value immediately followed by its `_unit` companion, all values defaulting to `NULL`. Positional calls `create_schema("S", params, items)` remain valid.

## Repo conventions

- Base pipe / no `%>%`: no `%>%` introduced; `%||%` used at `R/create_schema.R:150,177` is the rlang null-coalescing operator already used pervasively across the package and available in the namespace (rlang is in `Imports`). Pass.
- Exported-before-internal ordering: `create_schema()` (exported) precedes the internal `check_finite_scalar()` helper (`R/create_schema.R:195`). Pass.
- Commit message: no `Co-Authored-By` trailer; per-file bullet structure. Pass.
- PR body: opens with `Closes #167`, then the build summary; no "Generated with Claude Code" footer. Pass.
- Only the five expected files changed (`git diff --stat origin/main...HEAD`): `NEWS.md`, `R/create_schema.R`, `man/create_schema.Rd`, `tests/testthat/_snaps/create_schema.md`, `tests/testthat/test-create_schema.R`. No unrelated code touched. Pass.

## Adversarial / edge-case findings

- The `LocalizedParameter`-vs-`Parameter` distinction (AC-12) was the highest-risk spot: `create_parameter(path=...)` returns a subclass, but it inherits from `Parameter`, so the `inherits(..., "Parameter")` gate still reads `$data`. Verified and covered by a recorded snapshot. No defect.
- `NA` path: `NA` is logical, `is.numeric(NA)` is `FALSE`, so it hits the finite-scalar message rather than erroring inside `round()`. Verified; snapshot present.
- Non-scalar `c(1,2)`: `check_finite_scalar` aborts on `length != 1` before the whole-number branch, so `round()` never runs on a vector. Correct ordering.
- Byte-identity for the `parameters`-only path: confirmed the merge logic does not double-normalize or wrap the existing path. No behavioural drift.

## Deviations from spec/plan

- The plan text says the NEWS bullet "sorts before `fraction_unbound()`"; the section had since also gained a `create_snapshot()` bullet, and the implementation places `create_schema()` first, which is the correct alphabetical position (`create_schema` < `create_snapshot`). Not a deviation in substance; the alphabetical ordering requirement (FR-12) is satisfied. Acceptable.
- The plan suggested a positivity-neutral `number_of_repetitions` (accept `0`/negatives, finite + whole only); the implementation matches this (no positivity floor). Acceptable, matches spec edge-case note.

## Known-unrelated context (not held against the PR)

The 3 pre-existing `test-snapshot.R` version-ceiling failures are independently reproduced on `origin/main` without this change and are unrelated to issue 167. The `create_schema` test file itself is green. Not a blocker.

## Verdict

All 16 acceptance criteria map to concrete, correct evidence in the diff and supporting sources; error wording is captured in recorded snapshots that match the implementation; conventions are respected; no correctness defects were found. Approve.

</details>

## Summary by CodeRabbit

* **New Features**
  * Added direct options for configuring repetition count, repetition intervals, and schema start time when creating schemas.
  * Supports custom time units, with hours used by default.
  * Existing parameter customization remains available for other schema settings.

* **Bug Fixes**
  * Added validation for invalid repetition values, time units, and malformed parameter entries.
  * Clear errors are now reported when the same setting is provided through multiple options.

### CODE REVIEW

File: R/create_schema.R [hunk 1]
```diff
@@ -40,18 +72,114 @@
...
+    )
+    conflicting <- intersect(promoted_names, existing_names)
+    if (length(conflicting) > 0) {
+      cli::cli_abort(c(
+        "A schema parameter was supplied both as a promoted argument and in {.arg parameters}.",
+        "i" = "Supply each repetition parameter either as a plain argument ({.arg number_of_repetitions}, {.arg time_between_repetitions}, {.arg start_time}) or as an entry in {.arg parameters}, not both.",
+        "x" = "Conflicting parameter{?s}: {.val {conflicting}}."
+      ))
+    }
+  }
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply R/create_schema.R:180]: Fixed in c39abc6. `to_raw_parameters(parameters, "Name")` is now hoisted above the conflict check and reused in the merge, and `existing_names` is derived from its normalised `$Name`. I also added the missing entry validation to `to_raw_parameters()` itself, so a non-Parameter, non-list entry (e.g. a plain string) now raises the friendly "Every entry of `parameters` must be a <Parameter> or a raw list" error instead of the base-R `$ operator is invalid for atomic vectors` crash. Covered by a new `expect_snapshot(error = TRUE)` test.

File: R/create_schema.R [hunk 2]
```diff
@@ -40,18 +72,114 @@
...
+      )
+    }
+    promoted <- c(
+      promoted,
+      list(create_parameter(
+        name = "NumberOfRepetitions",
+        value = number_of_repetitions
+      ))
+    )
+  }
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply R/create_schema.R:113]: Fixed in c39abc6. The `NumberOfRepetitions` parameter now stores `round(number_of_repetitions)`, so a value like `2.9999999999999996` that clears the near-whole-number check is persisted as the exact integer `3`. Added a test asserting the stored value is `3`.
